File: samples/VisualBasic/Northwind/NorthwindDatabase.vb
Imports DbExtensions

Namespace Northwind

   Public Class NorthwindDatabase
      Inherits Database

      ReadOnly Property Products As SqlTable(Of Product)
         Get
            Return Table(Of Product)()
         End Get
      End Property

      ReadOnly Property Orders As SqlTable(Of Order)
         Get
            Return Table(Of Order)()
         End Get
      End Property

      ReadOnly Property OrderDetails As SqlTable(Of OrderDetail)
         Get
            Return Table(Of OrderDetail)()
         End Get
      End Property

      ReadOnly Property Employees As SqlTable(Of Employee)
         Get
            Return Table(Of Employee)()
         End Get
      End Property

      ReadOnly Property EmployeeTerritories As SqlTable(Of EmployeeTerritory)
         Get
            Return Table(Of EmployeeTerritory)()
         End Get
      End Property

      ReadOnly Property Regions As SqlTable(Of Region)
         Get
            Return Table(Of Region)()
         End Get
      End Property

      Sub New(connectionString As String, providerInvariantName As String)
         MyBase.New(connectionString, providerInvariantName)
      End Sub

   End Class

End Namespace

File: samples/CSharp/Northwind/NorthwindDatabase.cs
using System;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   public class NorthwindDatabase : Database {

      public SqlTable<Product> Products => Table<Product>();

      public SqlTable<Order> Orders => Table<Order>();

      public SqlTable<OrderDetail> OrderDetails => Table<OrderDetail>();

      public SqlTable<Employee> Employees => Table<Employee>();

      public SqlTable<EmployeeTerritory> EmployeeTerritories => Table<EmployeeTerritory>();

      public SqlTable<Region> Regions => Table<Region>();

      public NorthwindDatabase(string connectionString, string providerInvariantName)
         : base(connectionString, providerInvariantName) { }
   }
}
